.github/workflows/build-push-tool-images.yml

```diff
@@ -0,0 +1,66 @@
+name: Build & Push Tool Image to GHCR
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    paths:
+      - '.github/workflows/build-push-tool-images.yml'
+      - 'src/loaders/compute_tools/**/Dockerfile'
+  push:
+    branches:
+      - main
+      - master
+      - develop
+    paths:
+      - '.github/workflows/build-push-tool-images.yml'
+      - 'src/loaders/compute_tools/**/Dockerfile'
+
+jobs:
+  build-push-tool-images:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out GitHub Repo
+        uses: actions/checkout@v3
+
+      - name: Get current date
+        id: date
+        run: echo "date=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_OUTPUT
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: '${{ secrets.GHCR_USERNAME }}'
+          password: '${{ secrets.GHCR_TOKEN }}'
+
+      - name: Build and push tool images
+        id: build-and-push
+        run: |
+          tools=(
+            mash
+            gtdb_tk
+            checkm2
+            microtrait)
+
+          for tool in "${tools[@]}"; do
+            dockerfile="./src/loaders/compute_tools/$tool/Dockerfile"
+            tag="ghcr.io/${{ github.event.repository.owner.login }}/${{ github.event.repository.name }}:$tool"
```

Comment:
Also once we start adding version tags to the repo those should be incorporated into the tool tags, I'd think

Reply:
I think `github.event.repository.name` will always be `collections`. Depends on how we add version tags to the main repo, this might need to be updated in the future. yes.

Reply:
Yeah the version would have to go into the image tag (and maybe the image metadata). 

Why not just figure it out now? Then we don't have to worry about it in the future. The way tagging works in other repos is that a manual trigger is required; it might make sense to do the same thing here

Reply:
we don't have a collection release version. I don't know what needs to be figure out here. I also didn't get why/how collection version needs to be tagged with tool image. How do you want checkm2 being tagged for collections:0.0.1 vs collections:0.0.2?

Reply:
All 3 of these comments are related to each other, I think, so I'll consolidate the conversation here.

The way I would do this is
* build all the tools every PR but don't push
    * This is relatively quick and is parallel to the tests, and makes sure the tool doesn't break
* Add an action for each tool, similar to the manual release action, that builds and pushes an image with a tag something like `<tool_name>_<githhash>`
    * When we need a new tool image we just manually activate that action, which probably isn't very often
* Add an action for each tool (unless they can be combined into the action above) that builds and pushes an image like the above but uses a tool specific release version, so we could tag the repo with `mash_0.1.0` or something and release from that.
    * This allows releasing new versions of tools without having to release a new version of the rest of the code / other tools, etc.
    * Maybe there's a way to parameterize the actions with the tool name in the actions UI so that we don't need a separate action for each tool
* By default the tool runner would just pull the latest version (and record the githash / version when we get to that), but it could have an argument for supplying specific tool tags.